src/Storages/MergeTree/MergeTreeDataMergerMutator.cpp

```diff
@@ -794,6 +794,12 @@ MergeTreeData::MutableDataPartPtr MergeTreeDataMergerMutator::mergePartsToTempor
     if (need_remove_expired_values)
         merged_stream = std::make_shared<TTLBlockInputStream>(merged_stream, data, new_data_part, time_of_merge, force_ttl);
 
+    if (data.hasSkipIndices())
+    {
+        merged_stream = std::make_shared<ExpressionBlockInputStream>(merged_stream, data.skip_indices_expr);
+        merged_stream = std::make_shared<MaterializingBlockInputStream>(merged_stream);
+    }
```

Comment:
But #11162 has been already reverted from 20.4 in d2ee7dc0f2ec617ff6dbec7658964d3e6e5368f3, this means that there is no test for this change right now, can you pick the test too then?